src/Microsoft.DotNet.XHarness.WebAssembly/xunit-runner.cs

```diff
@@ -0,0 +1,387 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Linq;
+
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+using Xunit.ConsoleClient;
+
+namespace Microsoft.DotNet.XHarness.WebAssembly
+{
+    class MsgBus : IMessageBus
+    {
+        public MsgBus(IMessageSink messageSink)
+        {
+            Sink = messageSink;
+        }
+
+        IMessageSink Sink { get; set; }
+
+        public bool QueueMessage(IMessageSinkMessage message)
+        {
+            return Sink.OnMessage(message);
+        }
+
+        public void Dispose()
+        {
+        }
+    }
+
+    class Discoverer : XunitTestFrameworkDiscoverer
```

Comment:
Can we move this class to be an IEnumerable and clean the code that is using the empty while loop? I think it will me more clear to move that 
```csharp
while (_discoverer.Step());
```
to a 
```csharp
foreach (var t in _discoverer.Tests) 
{
    // logic here
}
```

Even more clean if the logic is moved to a method and reduce the huge indentation. The compiler should be smart enough to write nice IL.

Reply:
I've refactored the runner a bit